### PR TITLE
`Development`: Refactor student actions buttons checks

### PR DESCRIPTION
**Pull request by tobias-lippert**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
### Motivation and Context

#6498 has already been merged to fix the bug but we want to improve the code quality a bit. That's why I opened this PR.
### Description
moved the checks if the team button and the start exercise buttons should be available into seperate methods to reduce code duplication and to avoid copy-paste errors. 
### Steps for Testing

Prerequisites:

1 Instructor
1 Students

1. Log in to Artemis
2. Navigate to Course Administration
3. Create a team exercise that is released (affected exercise types are file-upload, modeling and text)
4. navigate as a student to the exercise and verify no buttons are visible
5. add the student to a team
6. Navigate again as a student to the exercise and verify that both buttons are visible, clickable and no errors are produced when clicking them.

#### Performance Review
- [x] I confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2
### Test Coverage

 my changes are tested by the tests added in #6498

### CODE REVIEW

**Review by laadvo — COMMENTED** [review]
Tested on ts1, the buttons become visible and clickable when the student is added to a team. The "Start exercise" / "View submission" works, however when clicking the "Team" button nothing happens. I'm not sure what this button should do. No errors appear.

**Review by laadvo — APPROVED** [review]
Retested on ts1 with modeling, text and file upload exercises. The buttons work as expected

File: src/main/webapp/app/overview/exercise-details/exercise-details-student-actions.component.ts [hunk 1]
```diff
@@ -99,10 +99,18 @@ export class ExerciseDetailsStudentActionsComponent implements OnInit, OnChanges
     }
 
     /**
-     * Starting an exercise is not possible in the exam, otherwise see exercise.utils -> isStartExerciseAvailable
+     * Starting an exercise is not possible in the exam or if it's a team exercise and the student is not yet assigned a team, otherwise see exercise.utils -> isStartExerciseAvailable
      */
     isStartExerciseAvailable(): boolean {
-        return !this.examMode && isStartExerciseAvailable(this.exercise, this.gradedParticipation);
+        const individualExerciseOrTeamAssigned = (!this.exercise.teamMode || this.exercise.studentAssignedTeamId) as boolean;
```
Comment (JohannesStoehr) [inline src/main/webapp/app/overview/exercise-details/exercise-details-student-actions.component.ts:105]: I haven't seen `as boolean` in our code base yet. Usually I'd expect one of these two:
```suggestion
        const individualExerciseOrTeamAssigned = !!(!this.exercise.teamMode || this.exercise.studentAssignedTeamId);
```
```suggestion
        const individualExerciseOrTeamAssigned = (!this.exercise.teamMode || this.exercise.studentAssignedTeamId) ?? false;
```

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]
The problems with methods in the template is that they are sometimes invoked over and over again due to a poor change detection.
I'm not sure if there is an easy way to check this, if it is here the case. But that is usually the reason to avoid such methods

File: src/main/webapp/app/overview/exercise-details/exercise-details-student-actions.component.ts (hunk 1 above)
Comment (tobias-lippert) [reply src/main/webapp/app/overview/exercise-details/exercise-details-student-actions.component.ts]: I get your point but duplicating all the logic in four places isn't ideal either and let to the bug we fixed recently.

**Comment by simon-hng** [comment]
- There is a missing margin between the buttons and unintended styles. Remove `btn-group` class and add `gap-3` to the `div` surrounding both buttons
- Clicking `start exercise` and then `file upload` leads  to an `You are not authorized to access this page` error. Going back by to the previous page I see the `start exercise` button again

**Review by tunargs — CHANGES_REQUESTED** [review]
1. I also get `You are not authorized to access this page` error when clicking on upload button multiple times
2. Buttons are too close to each other

**Review by DominikRemo — COMMENTED** [review]
Tested on ts1 in testing session for file upload exercises. Reloading the exercise page and clicking on the upload button results sometimes, not always, in an error message (unauthorised)

**Review by terlan98 — APPROVED** [review]
Tested on TS2. The buttons are only visible when the user is in a team 👍

**Review by laadvo — APPROVED** [review]
Tested on ts2, both buttons are working. No errors appear.

**Review by JohannesStoehr — DISMISSED** [review]
Code changes look good to me

**Review by terlan98 — DISMISSED** [review]
Tested on TS1. Reapproved code and functionality.

**Review by terlan98 — APPROVED** [review]

**Review by danielbode — APPROVED** [review]
Code LGTM

**Review by JohannesStoehr — APPROVED** [review]
The variable is a even better solution than the method 👍

**Review by laadvo — APPROVED** [review]
Re-tested locally, buttons are working as expected

**Pull request merged**